python/ctsm/utils.py

```diff
@@ -191,6 +192,35 @@ def write_output(file, file_in, file_out, file_type):
     file.close()
 
 
+# Import 1-d latitude or longitude variable from a CESM history file (e.g., name lat or lon) and return it as a DataArray that can be used for writing CESM input files
+def import_coord_1d(ds, coordName):
+    da = ds[coordName]
+    if len(da.dims) != 1:
+        abort(f"Expected 1 dimension for {coordName}; found {len(da.dims)}: {da.dims}")
+    return da, len(da)
+
+
+# Import 1-d latitude or longitude variable from a CESM history file (e.g., name LATIXY or LONGXY) and return it as a 1-d DataArray that can be used as a coordinate for writing CESM input files
+def import_coord_2d(ds, coordName, varName):
+    da = ds[varName]
+    thisDim = [x for x in da.dims if coordName in x]
+    if len(thisDim) != 1:
+        abort(f"Expected 1 dimension name containing {coordName}; found {len(thisDim)}: {thisDim}")
```

Comment:
I made sure there was an error check in the tester for this and there is! So awesome.